src/Type/Php/JsonThrowOnErrorDynamicReturnTypeExtension.php

```diff
@@ -51,8 +71,15 @@ public function getTypeFromFunctionCall(
 		Scope $scope,
 	): Type
 	{
+		// update type based on JSON_THROW_ON_ERROR
```

Comment:
```suggestion
```
this comment is actually not true any more, is it? the extension would be doing more then, I'd even argue to rename it, but maybe in a follow-up?